core/src/main/java/org/infinispan/util/concurrent/BlockingTaskAwareExecutorServiceImpl.java

```diff
@@ -29,13 +29,16 @@
    private final BlockingQueue<BlockingRunnable> blockedTasks;
    private final ExecutorService executorService;
    private final TimeService timeService;
+   private final ControllerThread controllerThread;
    private volatile boolean shutdown;
 
-   public BlockingTaskAwareExecutorServiceImpl(ExecutorService executorService, TimeService timeService) {
-      this.blockedTasks = new LinkedBlockingQueue<BlockingRunnable>();
+   public BlockingTaskAwareExecutorServiceImpl(String controllerThreadName, ExecutorService executorService, TimeService timeService) {
+      this.blockedTasks = new LinkedBlockingQueue<>();
```

Comment:
I wonder is there a reason this is specifically a BlockingQueue?  We don't block waiting for elements to be added, we instead use the semaphore and we don't have a bounds (besides Integer.MAX_VALUE).  Can this not just be a standard ConcurrentLinkedQueue ?